### PR TITLE
Detach Ore Prefixes from Meta Items

### PR DESCRIPTION
**Pull request by htmlcsjs**
This PR detaches ore prefixes from meta items.
Consequently it expands the material ID limit to 32767.

Other changes are:
 - The consolidation of misc items into MetaItem1
 - The change of loot tables

### CODE REVIEW

File: src/main/java/gregtech/api/pipenet/block/material/BlockMaterialPipe.java [hunk 1]
```diff
@@ -41,7 +42,11 @@ public ItemStack getItem(PipeType pipeType, Material material) {
 
     @Override
     public PipeType getItemPipeType(ItemStack itemStack) {
-        return getPipeTypeClass().getEnumConstants()[itemStack.getMetadata() / 1000];
+        GTLog.logger.info(itemStack.getMetadata());
+        if (itemStack.getMetadata() < 32000) {
+            return getPipeTypeClass().getEnumConstants()[itemStack.getMetadata() / 1000];
```
Comment (TechLord22) [inline src/main/java/gregtech/api/pipenet/block/material/BlockMaterialPipe.java:47]: You probably do not want / 1000 here.

File: src/main/java/gregtech/api/unification/material/Materials.java [hunk 2]
```diff
@@ -613,6 +584,41 @@ public static void register() {
...
+    public static DustMaterial CobaltOxide = new DustMaterial(392, "cobalt_oxide", 0x788000, DULL, 1, of(new MaterialStack(Cobalt, 1), new MaterialStack(Oxygen, 1)), 0);
+    public static DustMaterial ArsenicTrioxide = new DustMaterial(393, "arsenic_trioxide", 0xFFFFFF, ROUGH, 1, of(new MaterialStack(Arsenic, 2), new MaterialStack(Oxygen, 3)),0);
+    public static DustMaterial Massicot = new DustMaterial(394, "massicot", 0xFFDD55, DULL, 1, of(new MaterialStack(Lead, 1), new MaterialStack(Oxygen, 1)), 0);
+    public static DustMaterial Ferrosilite = new DustMaterial(395, "ferrosilite", 0x97632A, DULL, 1, of(new MaterialStack(Iron, 1), new MaterialStack(Silicon, 1), new MaterialStack(Oxygen, 3)), 0);
+    public static DustMaterial MetalMixture = new DustMaterial(396, "metal_mixture", 0x502d16, METALLIC, 1, of(), DISABLE_DECOMPOSITION);
+    public static DustMaterial SodiumHydroxide = new DustMaterial(397, "sodium_hydroxide", 0x003380, DULL, 1, of(new MaterialStack(Sodium, 1), new MaterialStack(Oxygen, 1), new MaterialStack(Hydrogen, 1)), 0);
+
+
```
Comment (TechLord22) [inline src/main/java/gregtech/api/unification/material/Materials.java:621]: Please rearrange these IDs to the proper places.

File: src/main/java/gregtech/api/unification/ore/OrePrefix.java [hunk 3]
```diff
@@ -349,7 +349,7 @@ private static void excludeAllGems(Material material) {
 
     public byte maxStackSize = 64;
     public final List<MaterialStack> secondaryMaterials = new ArrayList<>();
-    public float heatDamage = 0.0F; // Negative for Frost Damage
+    public float heatDamage = 0.0F; // Negative for Frost Damage;
```
Comment (TechLord22) [inline src/main/java/gregtech/api/unification/ore/OrePrefix.java:352]: What's this change for?

**Review by TechLord22 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/unification/ore/OrePrefix.java (hunk 3 above)
Comment (htmlcsjs) [reply src/main/java/gregtech/api/unification/ore/OrePrefix.java:352]: Err, typo? Definitely unintended

File: src/main/java/gregtech/api/unification/material/Materials.java (hunk 2 above)
Comment (htmlcsjs) [reply src/main/java/gregtech/api/unification/material/Materials.java:621]: what do you mean?

File: src/main/java/gregtech/api/pipenet/block/material/BlockMaterialPipe.java (hunk 1 above)
Comment (htmlcsjs) [reply src/main/java/gregtech/api/pipenet/block/material/BlockMaterialPipe.java:47]: turns out removing it causes a crash

File: src/main/java/gregtech/api/unification/material/Materials.java (hunk 2 above)
Comment (htmlcsjs) [reply src/main/java/gregtech/api/unification/material/Materials.java:621]: ok, do i shift id's for that?

**Review by TechLord22 — APPROVED** [review]

File: src/main/java/gregtech/common/items/MetaItem1.java [hunk 4]
```diff
@@ -477,52 +659,127 @@ public void registerRecipes() {
...
-        Material material = Material.MATERIAL_REGISTRY.getObjectById(damage % 1000);
-        OrePrefix prefix = this.orePrefixes[(damage / 1000)];
-        if(prefix != null && material != null) {
-            boolean isSolidState = prefix == OrePrefix.ingot || prefix == OrePrefix.gem;
-            boolean isMaterialTiered = material instanceof SolidMaterial && ((SolidMaterial) material).harvestLevel >= 2;
-            return isSolidState && isMaterialTiered;
-        }
-        return false;
-    }
+        // Dyes recipes
```
Comment (Irgendwer01) [inline src/main/java/gregtech/common/items/MetaItem1.java:663]: Please move all those recipes to the recipe classes.

**Review by Irgendwer01 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/items/MetaItem1.java (hunk 4 above)
Comment (htmlcsjs) [reply src/main/java/gregtech/common/items/MetaItem1.java:663]: i think @DStrand1 wants to do a big refractor so ill wait for his input

**Review by brachy84 — APPROVED** [review]

**Review by DStrand1 — APPROVED** [review]

**Review by TechLord22 — APPROVED** [review]

**Pull request merged**